paper/src/main/java/vg/civcraft/mc/citadel/model/CitadelDAO.java

```diff
@@ -360,57 +449,77 @@ public void fill(TableBasedBlockChunkMeta<Reinforcement> chunkData, Consumer<Rei
 		int preMultipliedZ = chunkData.getChunkCoord().getZ() * 16;
 		ReinforcementTypeManager typeMan = Citadel.getInstance().getReinforcementTypeManager();
 		World world = chunkData.getChunkCoord().getWorld();
-		try (Connection insertConn = db.getConnection();
-				PreparedStatement selectRein = insertConn.prepareStatement(
-						"select x_offset, y, z_offset, type_id, group_id, creation_time, health, insecure "
-								+ "from ctdl_reinforcements where chunk_x = ? and chunk_z = ? and world_id = ?;");) {
-			selectRein.setInt(1, chunkData.getChunkCoord().getX());
-			selectRein.setInt(2, chunkData.getChunkCoord().getZ());
-			selectRein.setShort(3, chunkData.getChunkCoord().getWorldID());
-			try (ResultSet rs = selectRein.executeQuery()) {
-				while (rs.next()) {
-					int xOffset = rs.getByte(1);
-					int x = xOffset + preMultipliedX;
-					int y = rs.getShort(2);
-					int zOffset = rs.getByte(3);
-					int z = zOffset + preMultipliedZ;
-					Location location = new Location(world, x, y, z);
-					short typeID = rs.getShort(4);
-					ReinforcementType type = typeMan.getById(typeID);
-					if (type == null) {
-						logger.log(Level.SEVERE, "Failed to load reinforcement with type id " + typeID);
-						continue;
+
+		// Add to a list to avoid connection deadlocks.
+		List<Reinforcement> inserts = new ArrayList<>();
```

Comment:
Not sure we need to create this list here.
The consumer function is pretty fast, this is what the consumer calls in the end (isNew = false):
https://github.com/CivMC/CivModCore/blob/master/paper/src/main/java/vg/civcraft/mc/civmodcore/world/locations/chunkmeta/block/BlockBasedChunkMeta.java#L227